tests/Unit/UseCases/ApplyForMembership/MembershipApplicationBuilderTest.php

```diff
@@ -137,6 +137,9 @@ public function testWhenNoBirthDateAndPhoneNumberIsGiven_membershipApplicationIs
 		$application = ( new MembershipApplicationBuilder() )->newApplicationFromRequest( $request );
 
 		$this->assertIsExpectedCompanyPersonName( $application->getApplicant()->getName() );
+		$this->assertTrue( $application->getApplicant()->getName()->isCompany() );
+		$this->assertFalse( $application->getApplicant()->getName()->isPrivatePerson() );
```

Comment:
@gbirke These two new lines here should actually cover the new logic, the rest of the repository logic also already covered by existing tests from what I can tell.